### PR TITLE
Fix battle AI protecting paralyzed archers

### PR DESCRIPTION
**Pull request by idshibanov**
AI was counting disabled shooters skewing battle estimations the wrong way.

### CODE REVIEW

File: src/fheroes2/battle/battle_troop.cpp [hunk 1]
```diff
@@ -763,6 +763,11 @@ uint32_t Battle::Unit::Resurrect( const uint32_t points, const bool allow_overfl
     return resurrect;
 }
 
+bool Battle::Unit::canShoot() const
+{
+    return isArchers() && !Modes( SP_BLIND | SP_PARALYZE | SP_STONE ) && !isHandFighting();
+}
```
Comment (ihhub) [inline src/fheroes2/battle/battle_troop.cpp:769]: We can make this method inlined by moving the implementation into the header file.

**Review by ihhub — COMMENTED** [review]
Hi @idshibanov , I left one comment here. Would you mind to take a look when you have free time?

File: src/fheroes2/battle/battle_troop.cpp [hunk 2]
```diff
@@ -763,6 +763,11 @@ uint32_t Battle::Unit::Resurrect( const uint32_t points, const bool allow_overfl
     return resurrect;
 }
 
+bool Battle::Unit::canShoot() const
+{
+    return isArchers() && !Modes( SP_BLIND | SP_PARALYZE | SP_STONE ) && !isHandFighting();
```
Comment (oleg-derevenetz) [inline src/fheroes2/battle/battle_troop.cpp:768]: There is a "unifying" constant:

```suggestion
    return isArchers() && !Modes( SP_BLIND | IS_PARALYZE_MAGIC ) && !isHandFighting();
```

File: src/fheroes2/battle/battle_troop.cpp (hunk 1 above)
Comment (idshibanov) [reply src/fheroes2/battle/battle_troop.cpp:769]: I thought about it but it would require add additional include to battle_troop.h file. What is worse?

Comment (ihhub) [reply src/fheroes2/battle/battle_troop.cpp:769]: Let's leave it as it is then.

**Review by ihhub — APPROVED** [review]

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
@idshibanov , thank you so much for this improvement!

**Pull request merged**